props-mongodb/src/intTest/java/sh/props/mongodb/MongoDbStoreIntTest.java

```diff
@@ -0,0 +1,216 @@
+import static sh.props.mongodb.testfixtures.MongoDbFixtures.generateRandomAlphanum;
+import static sh.props.mongodb.testfixtures.MongoDbFixtures.getCollection;
+import static sh.props.mongodb.testfixtures.MongoDbFixtures.initClient;
+import static sh.props.mongodb.testfixtures.MongoDbFixtures.replicaSetPrimaryStepDown;
+
+import com.mongodb.MongoNamespace;
+import com.mongodb.MongoNotPrimaryException;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoCollection;
+import org.bson.Document;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.utility.DockerImageName;
+import sh.props.Registry;
+import sh.props.RegistryBuilder;
+
+@Testcontainers
+@SuppressWarnings("NullAway")
+class MongoDbStoreIntTest {
+  @Container
+  private static final MongoDBContainer mongoDBContainer =
+      new MongoDBContainer(DockerImageName.parse("mongo:5.0.4-focal")).withExposedPorts(27017);
+
+  private static final String PROPS = "props";
+  private MongoClient mongoClient;
+  private String connString;
+  private String dbName;
+  private MongoCollection<Document> collection;
+
+  @BeforeAll
+  static void beforeAll() {
+    mongoDBContainer.start();
+    assertThat(mongoDBContainer.isRunning(), equalTo(true));
+  }
+
+  @AfterAll
+  static void afterAll() {
+    mongoDBContainer.stop();
+  }
+
+  @BeforeEach
+  void setUp() {
+    // initialize the connection string and database name for this test
+    connString = connectionString(mongoDBContainer.getMappedPort(27017));
+    dbName = generateRandomAlphanum();
+    mongoClient = initClient(connString);
+
+    // create database prop(s)
+    collection = getCollection(mongoClient, dbName, PROPS);
+    collection.insertOne(createProp("my.prop", "value"));
+  }
+
+  @AfterEach
+  void tearDown() {
+    mongoClient.close();
+  }
+
+  @Test
+  void watchCollectionChangeStream() {
+    // ARRANGE
+    var source = new MongoDbStore(connString, dbName, PROPS, WATCH_CHANGE_STREAM);
+
+    // ACT
+    Registry registry = new RegistryBuilder(source).build();
+
+    // ASSERT
+    await().until(() -> registry.get("my.prop"), equalTo("value"));
+
+    collection.insertOne(createProp("my.prop2", "value"));
+    await().until(() -> registry.get("my.prop2"), equalTo("value"));
+
+    collection.replaceOne(createFilter("my.prop2"), createProp("my.prop2", "value2"));
+    await().until(() -> registry.get("my.prop2"), equalTo("value2"));
+  }
+
+  @Test
+  void reloadAllDataFromCollection() {
+    // ARRANGE
+    var source = new MongoDbStore(connString, dbName, PROPS, RELOAD_ON_DEMAND);
```

Comment:
happy case which reloads all the data from the DB, on updates (e.g. to be used with `Scheduler.refresh(Source, ...)`